lib/pure/collections/deques.nim

```diff
@@ -165,73 +219,185 @@ template default[T](t: typedesc[T]): T =
   v
 
 proc popFirst*[T](deq: var Deque[T]): T {.inline, discardable.} =
-  ## Remove and returns the first element of the `deq`.
+  ## Remove and returns the first element of the ``deq``.
   emptyCheck(deq)
   dec deq.count
   result = deq.data[deq.head]
   deq.data[deq.head] = default(type(result))
   deq.head = (deq.head + 1) and deq.mask
 
 proc popLast*[T](deq: var Deque[T]): T {.inline, discardable.} =
-  ## Remove and returns the last element of the `deq`.
+  ## Remove and returns the last element of the ``deq``.
   emptyCheck(deq)
   dec deq.count
   deq.tail = (deq.tail - 1) and deq.mask
   result = deq.data[deq.tail]
   deq.data[deq.tail] = default(type(result))
 
 proc `$`*[T](deq: Deque[T]): string =
-  ## Turn a deque into its string representation.
+  ## Return the string representation of ``deq``,
+  ## with the elements enclosed in square brackets ``[]``
   result = "["
   for x in deq:
     if result.len > 1: result.add(", ")
     result.add($x)
   result.add("]")
 
+proc `==`*[T](a, b: Deque[T]): bool {.inline.} =
+  ## Return ``true`` if the elements and their order match.
+  if a.count != b.count: return false
+  for i in 0..<a.len:
+    if a[i] != b[i]: return false
+  result = true
+
+
+proc toSeq*[T](deq: Deque[T]): seq[T] {.inline.} =
+  ## Return a the elements of ``deq`` as a sequence.
+  ##
+  ## Example:
+  ##
+  ## .. code-block:: nim
+  ##   var
+  ##     deq = @[1,2,3,4,5].toDeque
+  ##     sq = deq.toSeq
+  result = newSeq[T](deq.count)
+  var i = deq.head
+  for c in 0 ..< deq.count:
+    result[c] = deq.data[i]
+    i = (i + 1) and deq.mask
+
+proc toDeque*[T](s: seq[T]): Deque[T] {.inline.} =
+  ## Return the elements of sequence ``s`` as a ``Dequeu``.
+  ##
+  ## Example:
+  ##
+  ## .. code-block:: nim
+  ##   var
+  ##     deq = @[1,2,3,4,5].toDeque
+  let slen = s.len
+  result = initDeque[T](slen)
+  result.data.setLen(nextPowerOfTwo(slen))
+  for i,x in s.pairs:
+    result.data[i] = x
+  result.count = slen
+  result.tail = slen
+
+template newDequeWith*(len: int, init: untyped): untyped =
```

Comment:
``newDequeWith`` is ok with me. This time. We need to do something about the proliferation of trivialities for consistency reasons though.